lib/galaxy/model/__init__.py

```diff
@@ -186,6 +187,23 @@ def check_password( self, cleartext ):
         """
         return galaxy.security.passwords.check_password( cleartext, self.password )
 
+    def system_user_pwent(self, real_system_username):
+        '''Gives the system user pwent entry based on e-mail or username depending on the value in real_system_username'''
+        if real_system_username == 'user_email':
+            try:
+                system_user_pwent = pwd.getpwnam(self.email.split('@')[0])
+            except KeyError:
+                pass
+        elif real_system_username == 'username':
+            try:
+                system_user_pwent = pwd.getpwnam(self.username)
+            except KeyError:
+                pass
+        else:
+            log.warning("invalid configuration of real_system_username")
+            system_user_pwent = None
```

Comment:
You have to move this line at the start of the method, otherwise `system_user_pwent` will be undefined when `KeyError` exception is raised in the `if` or `elif` branches.